src/GraphQL.Tests/GraphQLValuesCacheTests.cs

```diff
@@ -13,7 +13,13 @@ public void GetInt()
         GraphQLValuesCache.GetInt("-1000000").ShouldBe(-1000000);
         GraphQLValuesCache.GetInt("1000000").ShouldBe(1000000);
 
-        Should.Throw<FormatException>(() => GraphQLValuesCache.GetInt("a")).Message.ShouldBe("Input string was not in a correct format.");
```

Comment:
It is unfortunate that MS has changed the FormatException message from .NET 6 -> .NET 7. Maybe the strict exception message check could be removed and just check for throwing `FormatException`. In that case the nasty ifdef could be eliminated.

Reply:
> It is unfortunate that MS has changed the FormatException message from .NET 6 -> .NET 7. 

Maybe yes, maybe no :) . Yes, I prefer to test exception messages, not only exception types. When I read the test code, I better understand the nature of the code tested. I would not call it a tough requirement, just a recommendation that I would line to adhere to.

> So we just update the tests - as you have already done - when necessary.

Yes. Update if necessary. Sometimes I use `StartsWith`/`Contains` instead of exact match.